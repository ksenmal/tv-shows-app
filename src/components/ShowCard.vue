<script setup>
import { RouterLink } from "vue-router";
import fn from "../utils/functions.js";

defineProps({
  item: Object,
});
</script>

<template>
  <RouterLink :to="`/show/${item.id}`">
    <div>
      <p>{{ item.name }}</p>
      <p v-if="item.rating" class="rating">
        {{ item.rating.average }}<span class="star">â˜…</span>
      </p>
    </div>
    <img v-if="item.image" :src="item.image.medium" :alt="item.name" />
    <p class="years">( {{ fn.getYear(item.premiered) }} )</p>
  </RouterLink>
</template>

<style scoped>
div {
  display: flex;
  justify-content: space-between;
  margin-bottom: 5px;
}
.star {
  color: var(--color-yellow);
}
.years {
  text-align: center;
}
</style>
