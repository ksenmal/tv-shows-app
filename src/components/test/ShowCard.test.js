import { shallowMount } from "@vue/test-utils";
import ShowCard from "../ShowCard.vue";
import { describe, expect, test } from "vitest";

describe("ShowCard suite", () => {
  test("Snapshot", () => {
    const wrapper = shallowMount(ShowCard, {
      props: {
        item: {
          id: 1234567,
          rating: {
            average: 5,
          },
          image: {
            medium: "https://example.com/img.jpg",
          },
        },
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
