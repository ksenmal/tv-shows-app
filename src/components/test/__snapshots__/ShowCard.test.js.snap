// Vitest Snapshot v1

exports[`ShowCard suite > Snapshot 1`] = `"<router-link-stub to=\\"/show/1234567\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\" data-v-3daf941c=\\"\\"></router-link-stub>"`;
